lib/bolt/plugin.rb

```diff
@@ -250,6 +250,44 @@ def by_name(plugin_name)
       end
     end
 
+    # Loads all plugins and returns a map of plugin names to hooks.
+    #
+    def list_plugins
+      load_all_plugins
+
+      hooks = KNOWN_HOOKS.map { |hook| [hook, {}] }.to_h
+
+      @plugins.sort.each do |name, plugin|
+        # Don't show the Puppet Connect plugin for now.
+        next if name == 'puppet_connect_data'
```

Comment:
Heh, I literally added 'Add private metadata for plugins' right after you demoed this to my list of hacks. Someone from Skeletor might be interested in taking that up.